### ISSUE TITLE
boolean-`if` on optional invalidly allowed at `comptime`

### ISSUE THREAD

**Issue by rohlem** [ziglang/zig#21742]
### Zig Version

0.14.0-dev.1924+bdd3bc056

### Steps to Reproduce and Observed Behavior

```zig
fn b(arg: ?bool) bool {
    return if (arg) true else false;
}
fn u(arg: ?u0) bool {
    return if (arg) true else false;
}
test {
    _ = comptime b(true); //passes (yields false)
    _ = comptime u(0); //passes (yields false)
    //_ = b(true); //correctly issues compile error `expected type 'bool', found '?bool'`
    //_ = u(0); //correctly issues compile error `expected type 'bool', found '?u0'`
}
```

### Expected Behavior

Optional-unwrapping `if` should always require a capture, as it does in runtime code.
IMO this should count as a miscompilation, since invalid code (where the intent is not clear) is processed (and yields an unintuitive result).

**Comment by mlugg** [comment]
This will be solved by #21422, which I will be able to merge soon.

**Comment by rohlem** [comment]
duplicate of https://github.com/ziglang/zig/issues/21405 (less specific title)
